src/main/java/org/terasology/logic/health/event/HealthChangedEvent.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.logic.health.event;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.Event;
+
+/**
+ * This event (or a subtype) is sent whenever health changes
+ *
+ */
+public class HealthChangedEvent implements Event {
+    private EntityRef instigator;
+    private int change;
+
+    public HealthChangedEvent(EntityRef instigator, int change) {
+        this.instigator = instigator;
+        this.change = change;
+    }
+
+    public EntityRef getInstigator() {
+        return instigator;
+    }
+
+    /**
+     * @return The amount by which health changed. This is capped, so if the entity received 9999 damage and only had 10 health, it will be -10.
```

Comment:
Is it though? This method just seems to blindly return the change - I don't see any capping here? It could happen elsewhere but then that'd make this javadoc somewhat presumptuous :-)

It also helps to still write out both statements separately - the method description and the return value. Sometimes that can be tricky to write in a meaningful way split into two fields but since they each go to a different place in displayed javadoc it is ideal to have both (IMHO anyway)

An alternative might be to throw a boolean into a single event that indicates whether or not it resulted in a full (or partial) heal

Reply:
I think it does show the changed amount because the capping is done not in the event but in the method calling the event.
https://github.com/Terasology/Health/pull/1/files#diff-e0c50392da47b79c5d45c9c66a5ec5feR112